forge-game/src/main/java/forge/game/spellability/AbilityManaPart.java

```diff
@@ -361,6 +361,10 @@ public boolean meetsManaRestrictions(final SpellAbility sa, String restrictions)
                 }
             }
 
+            if (restriction.equals("CantCastNonArtifactSpells")) {
```

Comment:
i don't know if new Restriction is needed or if that can be done via the isValid check below

for does it collide with the getInstanceFromSpellAbility check?

Reply:
This is my first foray into writing in new engine support, so I'm a little lost trying to divine the context of those two lines you're referring to. My rationale was to look at similar mana restrictions, such as "CantCastSpellsFrom," and just create an OR boolean that would return true if: the mana pays for a cost that isn't a spell; or, if it is a spell, it is an artifact spell. 

This approach seemed like the path of least resistance for me, not knowing much else about the file, but I would like to learn more if I can! How does this risk conflicting with the `getInstanceFromSpellAbility` check? How can this be done using the `isValid` check?

Reply:
@Suthro i think it works for now

@Northmoc and @Agetian might later look at this and make it better

the problem there is that other Mana that uses Restriction can't be used for Costs that is part of a Trigger (When something you may pay, when you do, do something else)

this one can be used for such trigger.